languages/javascript/exercises/concept/closures/.docs/instructions.md

```diff
@@ -0,0 +1,54 @@
+Your design company has primarily been working with CSS transformations to build webpages. After some discussion, a decision is made
+to start using JavaScript to perform some of the calculations dynamically. Some of your teammates are less experienced with JavaScript,
+so you decide to use a function closure to create reusable transformation for `{x, y}` coordinate pairs.
+
+## 1. Translate the coordinates
+
+Create a function which returns a function making use of a closure to perform a repeatable 2d translation of a coordinate pair.
+
+```javascript
+const moveCoordinatesRight2Px = translate2d(2, 0)
+const result = moveCoordinatesRight2Px(4, 8)
+// result => [6, 8]
+```
+
+## 2. Scale the coordinates
+
+Create a function which returns a function making use of a closure to perform a repeatable 2d scale of a coordinate pair.
+
+> For the purposes of this exercise, assume only positive scaling values.
+
+```javascript
+const doubleScale = scale2d(2, 2)
+const result = doubleScale(6, -3)
+// result => [12, -6]
+```
+
+## 3. Compose transformation functions
+
+Combine two transformation functions to perform a repeatable transformation. This is often called _function composition_, where the result of the first function _'f(x)'_ is used as the input to the second function _'g(x)'_.
+
+```javascript
+const moveCoordinatesRight2Px = translate2d(2, 0)
+const doubleCoordinates = scale2d(2, 2)
+const composedTransformations = composeTransformation(
+  moveCoordinatesRight2Px,
+  doubleCoordinates
+)
+const result = composedTransformations(0, 1)
+// result => [4, 2]
+```
+
+## 4. Save the results of functions
+
+Make a function which _memoizes_ which takes a function, then returns a new function which remembers the inputs to the supplied function so that the value can be "remembered".
```

Comment:
I think the start of the sentence is a bit odd. Is there a word missing?

```suggestion
Make a function which takes a function to _memoize_, then returns a new function which remembers the inputs to the supplied function so that the value can be "remembered".
```

Also, maybe be slightly more explicit about what "remembered" means? Maybe something like:

```suggestion
Make a function which takes a function to _memoize_, then returns a new function which remembers the inputs to the supplied function so that the return value can be "remembered" and only calculated once.
```

Reply:
Having read the example implementation, it looks like only the last result should be remember. Maybe make this explicit in the instructions?

Reply:
```suggestion
Implement the `memoizeTransform` function. It takes a function to _memoize_, then returns a new function which remembers the inputs to the supplied function so that the last return value can be "remembered" and only calculated once if it is called again with the same arguments.
```